table/block_based/block_based_table_builder.cc

```diff
@@ -1564,8 +1567,15 @@ void BlockBasedTableBuilder::WriteFilterBlock(
           rep_->filter_builder->Finish(filter_block_handle, &s, &filter_data);
       assert(s.ok() || s.IsIncomplete());
       rep_->props.filter_size += filter_content.size();
+      bool top_level_filter_block = false;
+      if (!rep_->filter_builder->IsBlockBased() && s.ok() &&
```

Comment:
I would put `s.ok()` first and `!rep_->filter_builder->IsBlockBased()` last for performance. (In case of partitioned filters, `s.ok()` will usually be false and allow skipping the other tests.) (This is a hot code path.)